addon_service/authentication.py

```diff
@@ -12,7 +12,8 @@ def authenticate(self, request: DrfRequest):
         _user_uri = osf.get_osf_user_uri(request)
         if _user_uri:
             UserReference.objects.get_or_create(user_uri=_user_uri)
-            request.session["user_reference_uri"] = _user_uri
+            if not request.session.get("user_reference_uri"):
+                request.session["user_reference_uri"] = _user_uri
```

Comment:
Do you need to get_or_create the UserReference if it's already on the session? Might be able to save a db call here if not.

Reply:
Probably you still do need to do that, as I think about it more.

Reply:
I tried to run the scenario in my mind, I think we do. Here a potential scenario immediately after deployment to prod:
- A logged out user logs into OSF, `user_reference_uri` is set on the shared session
- User (now authenticated) then makes a call to GV, at this point a `UserReference` instance is not yet created for this user on the GV side